cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1491,20 +1484,16 @@ private AppFabricServiceStatus stop(Id.Program identifier, @Nullable String runI
   }
 
   private void getRuns(HttpResponder responder, Id.Program programId, String status,
-                       long start, long end, int limit) {
+                       long start, long end, int limit) throws BadRequestException {
     try {
-      try {
-        ProgramRunStatus runStatus = (status == null) ? ProgramRunStatus.ALL :
-          ProgramRunStatus.valueOf(status.toUpperCase());
-        List<RunRecord> records =
-          Lists.transform(store.getRuns(programId, runStatus, start, end, limit), CONVERT_TO_RUN_RECORD);
-        responder.sendJson(HttpResponseStatus.OK, records);
-      } catch (IllegalArgumentException e) {
-        responder.sendString(HttpResponseStatus.BAD_REQUEST,
-                             "Supported options for status of runs are running/completed/failed");
-      }
-    } catch (SecurityException e) {
```

Comment:
We don't need to catch SecurityExcepetion here and handle it the HTTPExceptionHandler does it. We are removing it from all the methods in this class.